server/serializers.py

```diff
@@ -492,48 +498,66 @@ def jsonize_clientside_tab(tab: clientside.TabUpdate) -> Dict[str, Any]:
     return d
 
 
-def _localize_module_message(
-    message_id: str,
-    *,
-    default: Optional[str],
-    ctx: JsonizeModuleContext,
-    arguments={},
-    log_not_found=False,
+def _localize_module_spec_message(
+    message_path: str, ctx: JsonizeModuleContext, spec_value: str
 ) -> str:
-    """Search the module catalogs for the message with the given id and localize it.
+    """Search the module catalogs for the spec message with the given path and localize it.
     
-    If the message is not found or is incorrectly formatted, `default` is returned.
+    The path of the message is its translation key minus "_spec." at the beginning.
     
-    In addition, if `default` is empty, `default` is returned.
+    If the message is not found or is incorrectly formatted, `spec_value` is returned.
+    
+    In addition, if `spec_value` is empty, `spec_value` is returned.
     This is in order to make sure that module spec values not defined in the spec file
     cannot be overriden by message catalogs.
     """
-    if not default:
-        return default
+    if not spec_value:
+        return spec_value
 
     try:
-        localizer = MESSAGE_LOCALIZER_REGISTRY.for_module_id(ctx.module_id)
-    except KeyError:
-        logger.exception(f"Unsupported module as I18nMessage source: {ctx.module_id}")
-        return default
+        return _localize_module_message(
```

Comment:
Please don't implement this in terms of `_localize_module_message()`. Currently, `_localize_module_message()` is complex (it logs _some_ stuff but not _other_ stuff; it raises errors when it shouldn't) just to support reuse in this function. That's not a worthwhile compromise.

These two functions should have two different docstrings and two different sets of behavior. Each possible problem should be enumerated in the docstring and code.